### ISSUE TITLE
Allow running proxy without listeners

### ISSUE THREAD

**Issue by Foorack** [PaperMC/Waterfall#271]
Essentially removing this line:
https://github.com/SpigotMC/BungeeCord/blob/master/proxy/src/main/java/net/md_5/bungee/conf/Configuration.java#L94

**Comment by Janmm14** [comment]
Why?

**Comment by kashike** [comment]
![but why](https://i.kym-cdn.com/photos/images/newsfeed/000/976/824/913.gif)

**Comment by Foorack** [comment]
Haha. This can be useful when you only want to run a processing server which has to be online at all times while yet using as little resources as possible.

**Example 1:** Running an empty bungee with Buycraft, votifier or any scheduled plugin on it to execute commands. There is no reason for this processing bungee to be listening for connections.

**Example 2:** I am making a small web endpoint API which only returns a yaml list of all staff and their UUID's when accessed. This couldn't be a standalone web process as it needs the LuckPerms API and it's environment. But I am also trying to trim it down as much as possible.

---
**tl;dr:** I don't see any reason *not* to allow this. ¯\_(ツ)_/¯

**Comment by electronicboy** [comment]
Removing that line is a no, I can somewhat see some pros for not requiring a listener to be defined at startup in the config, mainly for reasons of relying on plugins to deal with that as opposed to requiring a config...

At the end of the day, Bungee and waterfall have a specific job: To serve as a proxy for the minecraft protocol, your usage examples fall outside of what ultimatelyis the goal of the software, and while I personally do not mind changes which support the goal of the project to provide a system that works for all, I don't really see a massive pro for this in terms of real-world usage, the software would still need to scream at such misconfiguration, nor is there any API to support what I would personally consider a valid reason for such a change

**Comment by mikroskeem** [comment]
You can implement dumb core by grabbing BungeeCord/Waterfall API and implementing everything either with dumb code or minimal implementation to keep your services running.

**Comment by Janmm14** [comment]
@Foorack either just use your own fork's builds, possibly even stripping away more unneeded features

or the elegant way: dont use luckperms api (or change it to not be plugin-only working) for that and write your own code with a main method

**Comment by Foorack** [comment]
@mikroskeem Thank you for the idea but at this point I have been suggested both to re-implement the whole bungee core and re-implementing the LuckPerms core to be able to run the LuckPerms API without bungee completely, neither which are feasable. I have 5 exams this month, a goal for this project was to be able to run a web server with **minimum-effort**. :stuck_out_tongue: 

edit: @Janmm14 Not using LuckPerms API at all is really not an option, because the only purpose of the Web API is to fetch data *from LuckPerms*.

**Comment by Foorack** [comment]
The current setup \*works*. Yes it is listening for players but player count is set to 0, and it is firewalled off inside a docker container. Not the most elegant but works. Doesn't look like this issue will go through so closing it.

**Comment by Janmm14** [comment]
@Foorack Running a web server with minimal effort is "install apache2/nginx"

**Comment by Foorack** [comment]
@Janmm14 That wouldn't return the required data. I obviously meant miminum-effort for something which works. If I just wanted a web server in the laziest way possible then I would just run python3 -m http.server but that wouldn't solve anything.
